Property or indexer 'IOpenApiMediaType.Example' cannot be assigned to -- it is read only

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

Latest dotnet SDK 10, running on macOS M1/ARM. Getting this build error.

net10.0/generated/Microsoft.AspNetCore.OpenApi.SourceGenerators/Microsoft.AspNetCore.OpenApi.SourceGenerators.XmlCommentGenerator/OpenApiXmlCommentSupport.generated.cs(399,41): error CS0200: Property or indexer 'IOpenApiMediaType.Example' cannot be assigned to -- it is read only

Happened after I upgraded this morning. This is the error of code in the generated file

 foreach (var mediaType in content)
 {
      mediaType.Example = jsonString.Parse();
  }




### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Exceptions (if any)

_No response_

### .NET Version

_No response_

### Anything else?

_No response_

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -179,8 +179,8 @@
     <XunitExtensibilityExecutionVersion>$(XunitVersion)</XunitExtensibilityExecutionVersion>
     <XUnitRunnerVisualStudioVersion>3.1.3</XUnitRunnerVisualStudioVersion>
     <MicrosoftDataSqlClientVersion>5.2.2</MicrosoftDataSqlClientVersion>
-    <MicrosoftOpenApiVersion>[2.7.5)</MicrosoftOpenApiVersion>
-    <MicrosoftOpenApiYamlReaderVersion>[2.7.5)</MicrosoftOpenApiYamlReaderVersion>
+    <MicrosoftOpenApiVersion>[2.7.5,3.0.0)</MicrosoftOpenApiVersion>
+    <MicrosoftOpenApiYamlReaderVersion>[2.7.5,3.0.0)</MicrosoftOpenApiYamlReaderVersion>
     <!-- dotnet tool versions (see also auto-updated DotnetEfVersion property). -->
     <DotnetDumpVersion>6.0.322601</DotnetDumpVersion>
     <DotnetServeVersion>1.10.93</DotnetServeVersion>

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -179,8 +179,8 @@
     <XunitExtensibilityExecutionVersion>$(XunitVersion)</XunitExtensibilityExecutionVersion>
     <XUnitRunnerVisualStudioVersion>3.1.3</XUnitRunnerVisualStudioVersion>
     <MicrosoftDataSqlClientVersion>5.2.2</MicrosoftDataSqlClientVersion>
-    <MicrosoftOpenApiVersion>2.7.5</MicrosoftOpenApiVersion>
-    <MicrosoftOpenApiYamlReaderVersion>2.7.5</MicrosoftOpenApiYamlReaderVersion>
+    <MicrosoftOpenApiVersion>[2.7.5)</MicrosoftOpenApiVersion>
+    <MicrosoftOpenApiYamlReaderVersion>[2.7.5)</MicrosoftOpenApiYamlReaderVersion>
     <!-- dotnet tool versions (see also auto-updated DotnetEfVersion property). -->
     <DotnetDumpVersion>6.0.322601</DotnetDumpVersion>
     <DotnetServeVersion>1.10.93</DotnetServeVersion>

